apps/tx-builder/src/components/Dashboard.tsx

```diff
@@ -108,39 +115,34 @@ const Dashboard = (): ReactElement => {
 
       {/* ABI or Address Input */}
       <StyledAddressInput
-        id={'address-or-ABI-input'}
-        name="addressOrAbi"
-        label="Enter Address, ENS Name or ABI"
+        id="address"
+        name="address"
+        label="Enter Address or ENS Name"
         hiddenLabel={false}
-        address={addressOrAbi}
+        address={address}
         showNetworkPrefix={!!chainInfo?.shortName}
         networkPrefix={chainInfo?.shortName}
-        error={!isValidAddressOrContract ? loadContractError : ''}
+        error={!contractHasMethods ? loadContractError : ''}
         showLoadingSpinner={isABILoading}
         getAddressFromDomain={getAddressFromDomain}
-        onChangeAddress={(addressOrAbi: string) => setAddressOrAbi(addressOrAbi)}
+        onChangeAddress={(address: string) => setAddress(address)}
         InputProps={{
-          endAdornment: isValidAddressOrContract && (
+          endAdornment: contractHasMethods && isValidAddress(address) && (
             <InputAdornment position="end">
               <CheckIconAddressAdornment />
             </InputAdornment>
           ),
         }}
       />
 
-      {/* ABI Warning */}
-      {isValidAddressOrContract && !contract && (
-        <StyledWarningText color="warning" size="lg">
-          No ABI found for this address
-        </StyledWarningText>
-      )}
+      <JsonField id={'abi'} name="abi" label="Enter ABI" value={abi} onChange={setAbi} />
```

Comment:
We are not showing errors in the JsonField. Do you think that we can add it?
![Captura de pantalla 2022-02-09 a las 11 19 07](https://user-images.githubusercontent.com/26763673/153178452-9a192c77-537b-495a-8b73-8ea42439ecbd.png)

As a suggestion something like:

```
function isValidJSON(value: string | undefined) {
  if (value) {
    try {
      JSON.parse(value);
    } catch {
      return false;
    }
  }

  return true;
}
```

```
      <JsonField
        id={'abi'}
        name="abi"
        label="Enter ABI"
        value={abi}
        onChange={setAbi}
        error={isValidJSON(abi) ? '' : 'Invalid JSON value'}
      />
```
![Captura de pantalla 2022-02-09 a las 11 31 23](https://user-images.githubusercontent.com/26763673/153180784-674d4a28-5ab9-424b-ae79-1112d3daf3ad.png)

Reply:
Updated this. Nice!